geopandas/sindex.py

```diff
@@ -556,6 +678,29 @@ def query_bulk(self, geometry, predicate=None, sort=False):
 
             return res
 
+        @doc(BaseSpatialIndex.nearest)
+        def nearest(self, geometry):
+            if not compat.PYGEOS_GE_010:
+                raise NotImplementedError("sindex.nearest requires pygeos >= 0.10")
+
+            if isinstance(geometry, np.ndarray):
+                pass
+            elif isinstance(geometry, BaseGeometry):
+                geometry = array._shapely_to_geom(geometry)
+            elif isinstance(geometry, list):
+                geometry = [
+                    array._shapely_to_geom(el) if isinstance(el, BaseGeometry) else el
+                    for el in geometry
+                ]
+            elif isinstance(geometry, geoseries.GeoSeries):
+                geometry = geometry.values.data
+            elif isinstance(geometry, array.GeometryArray):
+                geometry = geometry.data
+            else:
+                geometry = np.asarray(geometry)
+
+            return super().nearest(geometry)
```

Comment:
@tomalrussell what do you think about using `nearest_all` to better match existing behavior from rtree, specifically for returning multiple equidistant results?

Reply:
On balance, I would still plan to expose both `nearest` and `nearest_all`, and stick with `nearest` here.

If future geopandas continues to support the rtree backend with a consistent spatial index API, it would be possible to write both pygeos-style `nearest` and `nearest_all` around the rtree `nearest` interface.

Having `nearest` available to return a single "nearest" result is sometimes useful - the index code (as e.g. in pygeos) may do a little less work, and when the client code wants only a single result, it doesn't have to discard multiples either.

We could write a single `nearest` with an `all=False` keyword argument - but that diverges from rtree too, and makes the shape of the output arrays potentially depend on that parameter. (It might be easier to read documentation for one method rather than two though? I could see this as a viable option. It would strictly add functionality to what's in this PR if we do go for it later.)

Reply:
> I would still plan to expose both `nearest` and `nearest_all`, and stick with `nearest` here.

+1 on that

> We could write a single nearest with an all=False keyword argument

I would personally expose them as two separate methods exactly as they are in pygeos.